### PR TITLE
TandemFoil: T_max=10 — gradient clipping max_norm=1.0

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The current TandemFoil best (PR #2490, val_primary/surface_pressure_mae=75.59) uses lion optimizer with lr=3e-4 and T_max=10 cosine annealing. The Lion optimizer can produce large gradient updates that may cause instability in surface pressure prediction. Gradient clipping with max_norm=1.0 constrains the gradient norm during training, which can stabilize training dynamics and improve convergence to a better minimum — particularly useful with Lion which is known to take larger steps than AdamW.

## Instructions
Run the following command from `target/icml2026/`:

```bash
python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 10 --no-use-ema --model_slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999 --max-grad-norm 1.0 --wandb_group tandem-tmax10-regularization
```

Key changes from baseline:
- All standard TandemFoil flags preserved
- `--max-grad-norm 1.0` (NEW — gradient clipping at norm=1.0)
- `--wandb_group tandem-tmax10-regularization` (grouped with other regularization runs)

**Important:** TandemFoil is restricted to 3L/192d architecture only — do not change `--model-layers` or `--model-hidden-dim` from defaults.

If `--max-grad-norm` is not a recognized flag, check `train.py --help` for the correct flag name (may be `--grad-clip`, `--clip-grad-norm`, or similar). Report in your PR comment which flag name worked.

Report in your PR comment:
- val_primary/surface_pressure_mae (primary metric, lower is better)
- All surface pressure metrics in a table
- W&B run ID and link
- Whether training was noticeably more stable with gradient clipping

## Baseline
Current best TandemFoil result (PR #2490):
- val_primary/surface_pressure_mae: **75.59**
- Reproduce command:
```bash
cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine_t_max 10 --no-use-ema --model_slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999
```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2667 because ADVISOR: Closing — gradient clipping at 3L/192d/lr=3e-4 is outdated. Baseline now 45.07 via lr=2e-4. naruto, please pick up your next assignment..
